harvester/harvester
# Migratable nodes option

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**
https://github.com/harvester/harvester/issues/3032#issue-1422120838

**Solution:**
Filter migratable nodes through by VM network to choose from.

**Related Issue:**
https://github.com/harvester/harvester/issues/3032

**Test plan:**
1. Create a ClusterNetwork.
2. Create a Network Config(at least two nodes) for ClusterNetwork.
3. Create a VM Network with the ClusterNetwork.
4. Create a VM with the VM Network.
5. Get and choose a migratable node from API: https://harvester/v1/harvester/kubevirt.io.virtualmachines/default/{your_vm}?action=getMigratableNodes.
```
{
    "nodes": [
        "node2"
    ]
}
```
6. (Optional) verify failure, choose a node without the ClusterNetwork and migrate, and check the response for error.
![image](https://user-images.githubusercontent.com/3347460/215450352-32d2eac4-b8ed-4055-8cf9-5f63d49bb34e.png)
7. Migrate to a migratable node and wait for the VM migration to succeed.
